lib/rubygems/package/tar_reader/entry.rb

```diff
@@ -230,8 +226,8 @@ def seek(offset, whence = IO::SEEK_SET)
     end
 
     while pending > 0 do
-      size_read = @io.read([pending, 4096].min).size
-      raise UnexpectedEOF if @io.eof?
+      ret = @io.readpartial([pending, 4096].min)
+      size_read = ret.size
```

Comment:
Same problem here where we ignored the possible `nil` return from read.

I also removed UnexpectedEOF here. I think there's two problems: If the read above hit an EOF without reading any bytes, then `.size` would raise. If it _properly_ read to right up to the EOF, like if this was the last file in the archive, then this would raise when it should not. The only reason to raise here is if we hit a normal EOFError due to readpartial, so I just let readpartial handle raising.

We could remove the UnexpectedEOF constant also. It was only (never) used here, so I think it's safe to remove completely. Besides, all EOFErrors are unexpected. 🤷‍♂️